### PR TITLE
Feature/emit thread

### PR DESCRIPTION
**Pull request by khampton353**

### CODE REVIEW

**Comment by khampton353** [comment]
Ticket 10480
This takes announcing event out of the path were the are created by pushing them to a dedicated emit-thread. Also, only connections with registered listeners are examined for interest in each event, as opposed to examining every connection on every server.

**Comment by khampton353** [comment]
Since watchers come and go with some frequency (or will once the unregister becomes more automatic) I chose the hash table as a better structure to handle volatility.

File: src/rpc_service.c [hunk 1]
```diff
@@ -160,12 +173,20 @@ rpc_context_create(void)
...
 
 	if (context == NULL)
 		return;
 
 	g_thread_pool_free(context->rcx_threadpool, true, true);
 	rpc_instance_free(context->rcx_root);
+
+	item = g_malloc(sizeof (*item));
+        item->context = NULL;
+        g_async_queue_push(context->rcx_emit_queue, item);
```
Comment (jceel) [inline src/rpc_service.c]: Some indentation issues here

Comment (khampton353) [reply src/rpc_service.c]: Ah, thats what I get for mixing languages. I'll fix it!

**Pull request merged**